distributed/client.py

```diff
@@ -1268,12 +1265,8 @@ def _handle_error(self, exception=None):
 
             # Give the scheduler 'stream-closed' message 100ms to come through
             # This makes the shutdown slightly smoother and quieter
-            with ignoring(AttributeError, gen.TimeoutError):
-                await gen.with_timeout(
-                    timedelta(milliseconds=100),
-                    self._handle_scheduler_coroutine,
-                    quiet_exceptions=(CancelledError,),
-                )
+            with ignoring(AttributeError, CancelledError):
+                await asyncio.wait([self._handle_scheduler_coroutine], timeout=0.1)
```

Comment:
This just blocks on `_handle_scheduler_coroutine` until 0.1 seconds pass or the coroutine completes/is cancelled. Is that what's intended here? Why not cancel it here on timeout?

If this is correct, I find the following easier to read:

```python
await asyncio.wait_for(asyncio.shield(self._handle_scheduler_coroutine), 0.1)
```

Reply:
Using `wait_for` with an 0.1s timeout was causing timeout error test failures. I tried bumping the timeout up but was still getting non-zero failures at 30s. Then I noticed we were previously ignoring `gen.TimeoutError`s here, so assumed that timeouts were okay in some cases.

Reply:
Hmmm, ok. Either as it is now, or as I wrote above (and adding `asyncio.TimeoutError` to the `ignoring`) would be equivalent. I'm not sure which I find easier to read.

Reply:
I like using `wait_for` + `shield`. It seems more explicit about what's going on, which is good in this case